packages/app/src/systems/Account/machines/accountsMachine.tsx

```diff
@@ -333,14 +363,53 @@ export const accountsMachine = createMachine(
             },
           });
 
-          // Add account to the database
+          // set as active account
           account = await AccountService.setCurrentAccount({
             address: account.address.toString(),
           });
 
           return account;
         },
       }),
+      importAccount: FetchMachine.create<
+        AccountInputs['importAccount'],
+        MachineServices['importAccount']['data']
+      >({
+        showError: true,
+        maxAttempts: 1,
+        async fetch({ input }) {
+          if (!input?.privateKey.trim()) {
+            throw new Error('Private key cannot be empty');
+          }
+          if (!input?.name.trim()) {
+            throw new Error('Name cannot be empty');
+          }
+
+          // Add account to vault
+          const accountVault = await VaultService.createVault({
+            type: 'privateKey',
+            secret: input.privateKey,
+          });
```

Comment:
You don't need to create a new vault for each privateKey a vault of type privateKey can handle multiple different privateKeys.

Reply:
https://github.com/FuelLabs/fuels-wallet/issues/617